contrib/src/uuid.rs

```diff
@@ -0,0 +1,133 @@
+extern crate uuid as uuid_ext;
+
+use std::fmt;
+use std::str::FromStr;
+use std::ops::{Deref, DerefMut};
+use rocket::request::FromParam;
+
+/// The UUID type, which implements `FromParam`. This type allows you to accept
+/// values from the [Uuid](https://github.com/rust-lang-nursery/uuid) crate as
+/// a dynamic parameter in your request handlers.
+///
+/// # Usage
+///
+/// To use, add the `uuid` feature to the `rocket_contrib` dependencies section
+/// of your `Cargo.toml`:
+///
+/// ```toml,ignore
+/// [dependencies.rocket_contrib]
+/// version = "*"
+/// default-features = false
+/// features = ["uuid"]
+/// ```
+///
+/// The UUID type implements the Deref trait allowing you to access the
+/// underlying Uuid type.
+///
+/// ```rust,ignore
+/// #[get("/users/<id>")]
+/// fn user(id: UUID) -> String {
+///     // Use Deref to access the underlying Uuid type.
+///     expects_uuid(*id);
+///
+///     format!("We found: {}", id)
+/// }
+/// ```
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub struct UUID(uuid_ext::Uuid);
+
+impl UUID {
+    /// Consumes the UUID wrapper returning the underlying Uuid type.
+    ///
+    /// # Example
+    /// ```rust,ignore
+    /// # extern crate uuid;
+    /// # use rocket_contrib::UUID;
+    /// # use std::str::FromStr;
+    ///
+    /// let uuid_str = "c1aa1e3b-9614-4895-9ebd-705255fa5bc2";
+    /// let my_inner_uuid = UUID::from_str(uuid_str).unwrap().into_inner();
+    /// assert_eq!(uuid::Uuid::from_str(uuid_str), my_inner_uuid)
+    /// ```
+    pub fn into_inner(self) -> uuid_ext::Uuid {
+        self.0
+    }
+}
+
+impl fmt::Display for UUID {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl<'a> FromParam<'a> for UUID {
+    type Error = uuid_ext::ParseError;
+
+    fn from_param(p: &'a str) -> Result<UUID, Self::Error> {
+        p.parse()
+    }
+}
+
+impl FromStr for UUID {
+    type Err = uuid_ext::ParseError;
+
+    fn from_str(s: &str) -> Result<UUID, Self::Err> {
+        Ok(UUID(try!(s.parse())))
+    }
+}
+
+impl Deref for UUID {
+    type Target = uuid_ext::Uuid;
+
+    fn deref<'a>(&'a self) -> &'a Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for UUID {
+    fn deref_mut<'a>(&'a mut self) -> &'a mut uuid_ext::Uuid {
```

Comment:
Only one that I can see. `fn clone_from(&mut self, source: &Self)` It would be pretty silly to deref_mut and then clone_from though... :).